gateway/ln-gateway/src/lib.rs

```diff
@@ -442,40 +443,59 @@ impl Gateway {
+                        let scid_to_feds = self.scid_to_federation.read().await;
+                        let federation_id = scid_to_feds.get(&htlc_request.short_channel_id);
+                        // Just forward the HTLC if we do not have a federation that
+                        // corresponds to the short channel id
+                        if let Some(federation_id) = federation_id {
+                            let clients = self.clients.read().await;
+                            let client = clients.get(federation_id);
+                            // Just forward the HTLC if we do not have a client that
+                            // corresponds to the federation id
+                            if let Some(client) = client {
+                                let htlc = htlc_request.clone().try_into();
+                                if let Ok(htlc) = htlc {
+                                    match client.gateway_handle_intercepted_htlc(htlc).await {
+                                        Ok(_) => continue,
+                                        Err(e) => {
+                                            info!(
+                                                "Got error intercepting HTLC: {e:?}, will retry..."
+                                            )
+                                        }
+                                    }
+                                } else {
+                                    info!("Got no HTLC result")
+                                }
+                            } else {
+                                info!("Got no client result")
                             }
                         }
-                    }
-                }
 
-                let outcome = InterceptHtlcResponse {
-                    action: Some(Action::Forward(Forward {})),
-                    incoming_chan_id: htlc_request.incoming_chan_id,
-                    htlc_id: htlc_request.htlc_id,
-                };
+                        let outcome = InterceptHtlcResponse {
+                            action: Some(Action::Forward(Forward {})),
+                            incoming_chan_id: htlc_request.incoming_chan_id,
+                            htlc_id: htlc_request.htlc_id,
+                        };
 
-                if let Err(error) = lnrpc.complete_htlc(outcome).await {
-                    error!("Error sending HTLC response to lightning node: {error:?}");
+                        if let Err(error) = lnrpc.complete_htlc(outcome).await {
+                            error!("Error sending HTLC response to lightning node: {error:?}");
+                        }
+                    }
+                    other => {
+                        info!("Got {other:?} while handling HTLC stream, exiting from loop...");
+                        break;
+                    }
                 }
             }
+        } else {
+            info!("Gateway isn't in a running state")
```

Comment:
IMO this should be an error, we shouldn't be intercepting HTLCs if we're not in the `Running` state